libs/daohaus-connect-feature/src/HausConnectContext.tsx

```diff
@@ -0,0 +1,145 @@
+import { isValidNetwork } from '@daohaus/common-utilities';
+
+import {
+  createContext,
+  ReactNode,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import {
+  getModal,
+  handleConnectWallet,
+  handleSwitchNetwork,
+  isMetamaskProvider,
+  loadProfile,
+  loadWallet,
+} from './utils/contextHelpers';
+
+import {
+  defaultWalletValues,
+  supportedNetworks,
+  web3modalDefaults,
+} from './utils/defaults';
+
+import {
+  ModalEvents,
+  ModalOptions,
+  NetworkConfigs,
+  ProviderType,
+  UserProfile,
+  WalletStateType,
+} from './utils/types';
+
+export type UserConnectType = {
+  provider: ProviderType | null | undefined;
+  chainId: string | null | undefined;
+  address: string | null | undefined;
+  profile: UserProfile;
+  connectWallet: () => Promise<void>;
+  disconnect: () => void;
+  isConnecting: boolean;
+  isConnected: boolean;
+  isMetamask: boolean;
+  networks: NetworkConfigs;
+  switchNetwork: (chainId: string) => void;
+  isProfileLoading: boolean;
+  isDaoScope: boolean;
+  validNetwork: boolean;
+};
+
+export const HausConnectContext =
+  createContext<UserConnectType>(defaultWalletValues);
+
+type ConnectProviderProps = {
+  web3modalOptions?: ModalOptions;
+  networks?: NetworkConfigs;
+  children: ReactNode;
+  handleModalEvents?: ModalEvents;
+};
+
+export const HausConnectProvider = ({
+  web3modalOptions = web3modalDefaults,
+  children,
+  networks = supportedNetworks,
+  handleModalEvents,
+}: ConnectProviderProps) => {
+  const [isConnecting, setConnecting] = useState(false);
+  const [{ provider, chainId, address }, setWalletState] =
+    useState<WalletStateType>({});
+  const [profile, setProfile] = useState<UserProfile>(null);
+  const [isProfileLoading, setProfileLoading] = useState(false);
+  const isConnected = useMemo(
+    () => !!provider && !!address && !!chainId,
+    [provider, address, chainId]
+  );
+  const isMetamask = useMemo(() => isMetamaskProvider(provider), [provider]);
+  const validNetwork = useMemo(
+    () => !!chainId && isValidNetwork(chainId, networks),
+    [chainId, networks]
+  );
+  // TODO, detect if we're in DAOscope by looking at the url.
```

Comment:
No way of deriving DAO scope until we start doing work on Core-UI or DAO context